icu4c/source/test/intltest/ustrtest.cpp

```diff
@@ -2351,6 +2352,14 @@ void UnicodeStringTest::TestUnicodeStringInsertAppendToSelf() {
     assertEquals("", u"abbcdcde", str);
 }
 
+void UnicodeStringTest::TestLargeMemory() {
+    IcuTestErrorCode status(*this, "TestLargeMemory");
+    constexpr uint32_t len = 2147483643;
+    char16_t *buf = new char16_t[len];
```

Comment:
This causes a MSVC compile error:

```
ustrtest.cpp(2358,25): error C2148: total size of array must not exceed 0x7fffffff bytes
```

Reply:
Right, and it won't work on a 32-bit platform either.
I guess we need some conditional compilation, either collecting platforms where this does not work, or collecting some where it does.

How about
```
#if U_PLATFORM_IS_LINUX_BASED || U_PLATFORM_IS_DARWIN_BASED
...
#endif
```

... and also:
```
    if (buf == nullptr) { return; }
```